Cargo.toml

```diff
@@ -8,24 +8,30 @@ default = ["ethereum"]
 ethereum = ["ethers-core"]
 
 [dependencies]
-num = { version = "0.4.3" }
+num = { version = "=0.4.0" }
 num-traits = { version = "0.2.15", default-features = false }
 num-bigint = { version = "0.4.3", default-features = false, features = [
     "rand",
 ] }
-
+anyhow = "1.0.95"
+rust-witness = "0.1.2"
+witnesscalc_adapter = { git = "https://github.com/zkmopro/witnesscalc_adapter.git", package = "witnesscalc-adapter" }
 
 # ZKP Generation
+ark-circom = { git = "https://github.com/zkmopro/circom-compat.git", branch = "wasm-delete"}
 ark-crypto-primitives = { version = "0.5.0" }
-ark-ec = { version = "0.5.0", default-features = false, features = ["parallel"] }
-ark-ff = { version = "0.5.0", default-features = false, features = ["parallel", "asm"] }
-ark-std = { version = "0.5.0", default-features = false, features = ["parallel"] }
-ark-bn254 = { version = "0.5.0" }
-ark-groth16 = { version = "0.5.0", default-features = false, features = ["parallel"] }
-ark-poly = { version = "0.5.0", default-features = false, features = ["parallel"] }
-ark-relations = { version = "0.5.1", default-features = false }
-ark-serialize = { version = "0.5.0", default-features = false }
-ark-bls12-381 =  { version = "0.5.0" }
+ark-ec = { version = "0.4.1", default-features = false, features = ["parallel"] }
+ark-ff = { version = "0.4.0", default-features = false, features = ["parallel", "asm"] }
+ark-std = { version = "0.4.0", default-features = false, features = ["parallel"] }
+ark-bn254 = { version = "0.4.0" }
+ark-groth16 = { version = "0.4.0", default-features = false, features = ["parallel"] }
+ark-poly = { version = "0.4.0", default-features = false, features = ["parallel"] }
+ark-relations = { version = "0.4", default-features = false }
+ark-serialize = { version = "0.4.0", default-features = false }
+ark-bls12-381 =  { version = "0.4.0" }
```

Comment:
can we upgrade the version?
or it fails to upgrade?

Reply:
No, we can't because of `ark-circom`, we're using our own branch 
`ark-circom = { git = "https://github.com/zkmopro/circom-compat.git", version = "0.1.0", branch = "wasm-delete", optional = true }`